http.go

```diff
@@ -94,134 +83,254 @@ const helpMessage = `usage: http [flag...] [METHOD] URL [REQUEST_ITEM [REQUEST_I
+	if err != nil {
+		fatalf("%v", err)
+	}
+	defer resp.Body.Close()
+	if err := showResponse(p, resp, os.Stdout); err != nil {
+		fatalf("%v", err)
+	}
+}
 
-	flag.BoolVar(&flags.form, "f", false, "serialize data items as form values")
-	flag.BoolVar(&flags.form, "form", false, "")
+func newContext(fset *flag.FlagSet, args []string) (*context, *params, error) {
+	p, err := parseArgs(fset, args)
+	if err != nil {
+		return nil, nil, err
+	}
+	if p.debug {
+		loggo.ConfigureLoggers("DEBUG")
+	}
+	ctxt := &context{
+		url:       p.url,
+		method:    p.method,
+		header:    make(http.Header),
+		urlValues: make(url.Values),
+		form:      make(url.Values),
+		jsonObj:   make(map[string]interface{}),
+	}
+	for _, kv := range p.keyVals {
+		if err := ctxt.addKeyVal(p, kv); err != nil {
+			return nil, nil, err
+		}
+	}
+	if p.basicAuth != "" {
+		ctxt.header.Set("Authorization",
+			"Basic "+base64.StdEncoding.EncodeToString([]byte(p.basicAuth)))
+	}
+	if p.json {
+		ctxt.header.Set("Content-Type", "application/json")
+	}
+	return ctxt, p, nil
+}
+
+func parseArgs(fset *flag.FlagSet, args []string) (*params, error) {
+	var p params
+	var printHeaders, noBody bool
+	fset.BoolVar(&p.json, "j", false, "serialize  data  items  as a JSON object")
+	fset.BoolVar(&p.json, "json", false, "")
+
+	fset.BoolVar(&p.form, "f", false, "serialize data items as form values")
+	fset.BoolVar(&p.form, "form", false, "")
 
-	flag.BoolVar(&flags.headers, "t", false, "print only the response headers")
-	flag.BoolVar(&flags.headers, "headers", false, "")
+	fset.BoolVar(&printHeaders, "h", false, "print the response headers")
+	fset.BoolVar(&printHeaders, "headers", false, "")
 
-	flag.BoolVar(&flags.body, "b", false, "print only the response body")
-	flag.BoolVar(&flags.body, "body", false, "")
+	fset.BoolVar(&noBody, "B", false, "do not print response body")
+	fset.BoolVar(&noBody, "body", false, "")
 
-	flag.BoolVar(&flags.noBrowser, "B", false, "do not open URLs in web browser")
-	flag.BoolVar(&flags.noBrowser, "no-browser", false, "")
+	fset.BoolVar(&p.debug, "debug", false, "print debugging messages")
 
-	flag.BoolVar(&flags.raw, "raw", false, "print response body without any JSON post-processing")
+	fset.BoolVar(&p.noBrowser, "W", false, "do not open macaroon-login URLs in web browser")
+	fset.BoolVar(&p.noBrowser, "no-browser", false, "")
 
-	flag.StringVar(&flags.basicAuth, "a", "", "http basic auth (username:password)")
-	flag.StringVar(&flags.basicAuth, "auth", "", "")
+	fset.BoolVar(&p.raw, "raw", false, "print response body without any JSON post-processing")
 
-	flag.StringVar(&flags.cookieFile, "cookiefile", filepath.Join(os.Getenv("HOME"), ".go-cookies"), "file to store persistent cookies in")
+	fset.StringVar(&p.basicAuth, "a", "", "http basic auth (username:password)")
+	fset.StringVar(&p.basicAuth, "auth", "", "")
+
+	fset.StringVar(&p.cookieFile, "cookiefile", filepath.Join(os.Getenv("HOME"), ".go-cookies"), "file to store persistent cookies in")
```

Comment:
If XDG_DATA_HOME was not platform-specific, I'd be tempted to agree. but it's not even set for me under ubuntu linux.

One possibility might be to require that a GO_COOKIEFILE environment variable be set; if it's not, we'll print a warning and use in-memory cookies.